src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationRecordPattern.java

```diff
@@ -0,0 +1,100 @@
+//non-compiled with javac: Compilable with Java21                           //indent:0 exp:0
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;     //indent:0 exp:0
+
+
+/* Config:                                                                  //indent:0 exp:0
+ * basicOffset = 4                                                          //indent:1 exp:1
+ * braceAdjustment = 0                                                      //indent:1 exp:1
+ * caseIndent = 4                                                           //indent:1 exp:1
+ * tabWidth = 4                                                             //indent:1 exp:1
+ * throwsIndent = 8                                                         //indent:1 exp:1
+ * forceStrictCondition = true                                              //indent:1 exp:1
+ */                                                                         //indent:1 exp:1
+public class InputIndentationRecordPattern {  //indent:0 exp:0
+    record ColoredPoint(boolean p, int x, int c) { } //indent:4 exp:4
+    record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { } //indent:4 exp:4
+
+    void test(Object obj) { //indent:4 exp:4
+        boolean b = obj instanceof //indent:8 exp:8
+                ColoredPoint(_,_,_);  //indent:16 exp:12 warn
+        boolean c = obj instanceof //indent:8 exp:8
+            ColoredPoint(_,_,_);  //indent:12 exp:12
+
+        boolean bb = obj instanceof //indent:8 exp:8
+        ColoredPoint(_,_,_);  //indent:8 exp:12 warn
```

Comment:
looks good to me

Reply:
What do you mean specifically about a line with a warning looking "good"?

Reply:
> this comment encompassing all 3 examples?

Yes I wanted to say that the check behaviour is good with these 3 examples and I see no issues

Reply:
I am use to always seeing context here even if just discussing 1 line, so I didn't realize you were talking about the entire context at first.